### DISCUSSION TITLE
Bref + Workerman/Webman or how to run the php start.php start command

### DISCUSSION THREAD

**Discussion by felipoantonoff** [brefphp/bref#1108] [Support]
Hi, would you know any way to run Bref + Workerman/Webman or how to run the php start.php start -d command.

I'm looking at using Workerman https://github.com/walkor/Workerman and the Webman Framework based on it: https://github.com/walkor/webman to be used on Lambda AWS and also on Cloud.

Locally it's very easy to run them, just run: `php start.php start -d`, but I don't know how you would do it in the case of Bref + Lambda AWS, maybe some way to instantiate Lambda AWS to run the command together.

The interesting thing about Workerman is that unlike Swoole it doesn't need to install as an extension but the simple use of composer, it seems a great way out to be used with Breb + Lambda AWS and also in direct Cloud.

Another excellent Framework to use with Lambda AWS is Flight PHP which is first rate, but Wokerman's advantage is its features similar to Swoole and with very good performance, having more options to use.

My idea is to keep a general base for both Cloud and Lambda AWS, so it has to have excellent performance in Serverless and also in Cloud.

**Comment by felipoantonoff** [comment]
Hello, unfortunately I have not had feedback on.

Should I use Flight PHP or Opencart same as the Framework, so I can easily run the same code on AWS Lambda and Cloud.

Now it would be interesting to have native support for Workerman/Webman by Bref and greater focus also on some microframework, such as Flight PHP itself, which has a very good performance and low consumption of resources.

**Reply by t-richard** [reply]
Hi!

I think you didn't get an answer because those frameworks are a bit niche so fewer people are able to help compared to Laravel or Symfony.

I don't know how those work but I guess you would need a bridge to transform the lambda request to a format your framework of choice understands because they seem to not work with php-fpm. 

You would not be able to run the command the given command because this is likely a long running process that is incompatible to how Lambda works.

If those have support for psr-7/psr-15 requests, you could use the native Bref support for those and forward the request to your framework.
See https://bref.sh/docs/function/handlers.html#api-gateway-http-events

If you need more help on a precise issue/question or if that wasn't clear, feel free to ask here again and please report back if you find a way to make it work for the next person that want to do the same thing 😊

**Comment by felipoantonoff** [comment]
Hi, thanks for the reply.

Yes, they are more niche, but I think it would be interesting for Bref and even for Cloud use also direct, I consider Laravel a little bloated even more for Serverless, so I think of Flight PHP or Webman.

Webman uses a simple command in the terminal, they don't really use PHP FPM, it is already the direct server similar to Swoole but without the dependency of installing by PHP extension.

That's right, the command needs to keep running until the function code finishes running.

Thanks for the tips and ideas, for now I think I'll use Flight PHP anyway, because for my use it should serve very well by Serverless and Cloud and I already have the base structure made in it that I used in the Bref and Serverless studies.

After launching the service, I return to inform you about my experience and the choices I made.

**Comment by wxfjamdc** [comment]
Hi.

I think it can be achieved through docker image.
